lib/Doctrine/DBAL/DBALException.php

```diff
@@ -77,10 +77,22 @@ public static function invalidPdoInstance()
     }
 
     /**
+     * @param bool|null $url The URL that was provided in the connection parameters (if any).
+     *
      * @return \Doctrine\DBAL\DBALException
      */
-    public static function driverRequired()
+    public static function driverRequired($url = null)
     {
+        if ($url) {
+            return new self(
+                sprintf(
+                    "The options 'driver' or 'driverClass' are mandatory if a connection URL without scheme " .
+                    "is given to DriverManager::getConnection(). Given URL: %s",
+                    $url
+                )
+            );
+        }
```

Comment:
I'd put this into a separate static method. To me only looking at this code it is not clear, that when `$url` is given here, it misses one special part. Maybe `missingSchemeRequiredDriver()` or something like that?